lib/phoenix_live_view/test/live_view_test.ex

```diff
@@ -1034,35 +1034,57 @@ defmodule Phoenix.LiveViewTest do
     call(view, :page_title)
   end
 
+
   @doc """
-  Asserts a live patch will happen within `timeout`.
+  Asserts a live patch will happen, optionally providing a `to` path
+  and a `timeout` in milliseconds. If a `to` path is not provided, will
+  assert true for any patch.
+
+  It returns the new path.
 
-  It always returns `:ok`. To assert on the flash message,
-  you can assert on the result of the rendered LiveView.
+  To assert on the flash message, you can assert on the result of the
+  rendered LiveView. to assert on the path, for example using a regular
+  expression, you can assert on the result of `path/1`.
 
   ## Examples
 
+      render_click(view, :event_that_triggers_patch)
+      assert_patch view
+
+      render_click(view, :event_that_triggers_patch)
+      path = assert_patch view
+      assert path =~ ~r/path/\d+/
+
       render_click(view, :event_that_triggers_patch)
       assert_patch view, "/path"
 
+      render_click(view, :event_that_triggers_patch)
+      assert_patch view, "/path", 30
+
   """
-  def assert_patch(%View{} = view, to, timeout \\ 100)
-      when is_binary(to) and is_integer(timeout) do
-    assert_navigation(view, :patch, to, timeout)
-    :ok
+  def assert_patch(view, to \\ nil, timeout \\ 100)
+    when (is_binary(to) or is_nil(to)) and is_integer(timeout) do
+
+    # Flash is always empty here as the flash isn't sent with a patch.
```

Comment:
@chrismccord is there a reason why there is no flash when asserting on patch?

Reply:
@chrismccord can answer this better, but when @cthulahoops and I dug into the code we got to here: https://github.com/phoenixframework/phoenix_live_view/blob/4c9f46d566d4f2c2062b68a1425faba8245d2595/lib/phoenix_live_view/channel.ex#L609-L655

It looks like on redirects and live redirects we call `copy_flash` which puts the flash into the the reply sent to the client, but on patches we don't do this. This makes sense, as when we redirect the browser puts the flash into a cookie (https://github.com/phoenixframework/phoenix_live_view/blob/1cad4ceb50b8417ecacdae0c0cbcc627ca71ce9e/assets/js/phoenix_live_view.js#L1381) which can then be picked up by the next render. If you're doing a live patch, the flash just lives in the socket's private store.

When it comes to the test code, it's easy to pick up the flash on redirects since we get it in the reply that's picked up by the client proxy. To get the flash on a patch we'd need to reach into the LiveView socket to get this :(. I think that when writing tests I *do* want to assert on the flash after a patch, but there doesn't seem to be a way to get this out cleanly with the way things are architected now... but maybe I'm missing something?

Reply:
Ah, I see! Yes, you are correct. With `assert_patch`, you can assert the flash by simply checking if it is currently showing on the page. So we need to decide if we want to address this and make the flash available for assert_patch, otherwise we shouldn't return the flash at all. In this case:

    assert_patch/1 #=> returns the path
    assert_patch/2 #=> returns :ok
    assert_redirect/1 #=> returns {path, flash}
    assert_redirect/2 #=> returns flash

Reply:
That seems reasonable to me. It's a bit unfortunate that they all have different shaped returns, but they are different functions.

Reply:
@josevalim We've updated the PR so all the return values are as suggested here. How does this look now?